.github/workflows/doctum.yml

```diff
@@ -0,0 +1,114 @@
+name: Generate Reference Documentation
+
+on:
+    workflow_call:
+        inputs:
+            title:
+                type: string
+                default: "Reference Documentation"
+                required: true
+            theme:
+                type: string
+                description: "doctum theme"
+                default: default
+            default_version:
+                type: string
+                description: "The version tag to use as the latest version."
+            tag_pattern:
+                type: string
+                description: "tags to include in version selector"
+                default: "v1.*"
+            dry_run:
+                type: boolean
+                description: "do not deploy to gh-pages"
+            exclude_file:
+              type: string
+              description: "exclude a file from documentation"
+
+permissions:
+  contents: write
+
+jobs:
+    docs:
+        name: "Generate and Deploy Documentation"
+        runs-on: ubuntu-latest
+        steps:
+        - name: Checkout
+          uses: actions/checkout@v4
+          with:
+            fetch-depth: 0
+        - name: Setup PHP
+          uses: shivammathur/setup-php@v2
+          with:
+            php-version: 8.1
+        - name: Download Doctum
+          run: |
+            curl -# https://doctum.long-term.support/releases/5.5/doctum.phar -o doctum.phar
+            curl -# https://doctum.long-term.support/releases/5.5/doctum.phar.sha256  -o doctum.phar.sha256
+            sha256sum --strict --check doctum.phar.sha256
+        - name: Generate Doctum Config
+          run: |
+            DOCTUM_CONFIG=$(cat <<'EOF'
+            <?php
```

Comment:
Can we use the contents of a file for this section instead of writing code inside the YAML file? Is that possible?

Reply:
With the shared workflows it isn't possible. At least as far as I know. That's because the repo isn't actually cloned - github just LOOKS at the shared workflow.

We could have the shared workflow clone this repo to get the file, or something along those lines, but I figured that it's simple enough that it's okay to just put it in here. But I'm definitely open to other solutions.

Reply:
Thanks Brent!